src/backend/app/Models/PaymentHistory.php

```diff
@@ -36,7 +36,7 @@ public function transaction(): BelongsTo {
     public function getFlow(string $payer_type, int $payer_id, string $period, $limit = null, string $order = 'ASC') {
         $sql = 'SELECT sum(amount) as amount, payment_type, CONCAT_WS("/", '.$period.') as aperiod from'.
             ' payment_histories where payer_id=:payer_id and payer_type=:payer_type '.
-            'GROUP by concat( '.$period.'), payment_type ORDER BY created_at  '.$order;
+            'GROUP by aperiod, payment_type ORDER BY aperiod  '.$order;
```

Comment:
I don't think we should change the `ORDER BY` clause here. Because then the result is weird. For example:

<img width="267" alt="image" src="https://github.com/user-attachments/assets/d73d9056-1547-4e8b-befd-9514f2c29201" />

It get's ordered lexiographically. And then it doesn't match a timeline

<img width="372" alt="image" src="https://github.com/user-attachments/assets/97694f1d-12e8-458e-be4d-32803078a893" />

(10-2023 appearing in the middle of 2024)

Reply:
Good catch! Had to remove `created_at` to get the query to work. Now that I understand this better I think I could get this work.

Reply:
Yeah, the error is very misleading in this situation. The `ORDER BY created_at` has nothing to do with the error, even though the error suggest it 😛.

In fact, if you remove the `ORDER BY created_at` entirely, the error still persists. 

It's only about the `GROUP BY` clause.

Reply:
Updated to order by created_at, the bar chart is now orderly correctly. 


<img width="578" alt="Screenshot 2024-12-28 at 8 18 29 PM" src="https://github.com/user-attachments/assets/c7258ae0-3266-4ca8-88ae-ebc2aaf4355f" />